src/components/implementation/tests/micro_booter/mb_tests.c

```diff
@@ -925,15 +876,35 @@ test_run_mb(void)
  * TODO: Fix those eventually.
  */
 void
+__block_vm(void)
+{
+	int blocked, rcvd;
+	cycles_t cycles, now;
+	tcap_time_t timeout, thd_timeout;
+	thdid_t tid;
+
+	while (cos_sched_rcv(BOOT_CAPTBL_SELF_INITRCV_BASE, RCV_ALL_PENDING, 0, &rcvd, &tid, &blocked, &cycles, &thd_timeout) > 0); 
```

Comment:
This is to receive any pending events, thanks for pointing out. I should also use `RCV_NON_BLOCKING` flag. The `cos_sched_rcv` after this line is guaranteed to block with a timeout.